### DISCUSSION TITLE
Use fastapi-users as an OIDC provider

### DISCUSSION THREAD

**Discussion by junoriosity** [fastapi-users/fastapi-users#912] [Q&A]
I am currently trying to install Kubeapps (https://github.com/kubeapps/kubeapps) and for this there is the possibility to use an OIDC provider for authentication (see https://github.com/kubeapps/kubeapps/blob/main/docs/user/using-an-OIDC-provider.md resp. https://github.com/kubeapps/kubeapps/blob/main/docs/user/using-an-OIDC-provider-with-pinniped.md).

Now that I have used fastapi-users so nicely for my fastAPI server, I was wondering whether we could somehow use/configure it for OIDC, such that this serves as the identity provider for my kubeapps here?

**Comment by frankie567** [comment]
Hi @junoriosity!

I definitely understand the need here. However, implementing an OIDC provider is complex and IMO, a bit out of the scope of this library. There are lots of things to take care of, especially the authorization code flow grant, managing OAuth2 clients, granted scopes, etc.

I know it because I'm actually working on a quite ambitious project aiming at providing an all-in-one authentication service. Currently, it's a bit early to say if I'll be able to pull some parts of it and provide them as FastAPI libraries, but that's a possibility.
